pkg/sync/common/types.go

```diff
@@ -50,6 +50,8 @@ const (
 	SyncOptionClientSideApplyMigration = "ClientSideApplyMigration=true"
 	// Sync option that disables client-side apply migration
 	SyncOptionDisableClientSideApplyMigration = "ClientSideApplyMigration=false"
+	// Sync option that means only controller owner references are respected
+	SyncOptionControllerReferencesOnly = "ControllerReferencesOnly=true"
```

Comment:
I'm curious how this options can become the default. Is there a concept in the Argo project as feature gates.

eg. where it's opt-in and then the flag get's switched to opt out?